lib/src/component/pure_ui_component.dart

```diff
@@ -0,0 +1,68 @@
+// Copyright 2020 Workiva Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import 'package:collection/collection.dart';
+import 'package:meta/meta.dart';
+import 'package:over_react/over_react.dart';
+
+/// A mixin to make a [UiComponent2] instance behave
+/// like a [ReactJS `PureComponent`](https://reactjs.org/docs/react-api.html#reactpurecomponent).
+///
+/// If your component does not support `props.children`, you can override [supportsPropChildren]
+/// to `false` for further optimizations.
+mixin PureUiComponent<T extends UiProps> on UiComponent2<T> {
+  @mustCallSuper
+  @override
+  bool shouldComponentUpdate(Map nextProps, Map nextState) {
+    final currentPropsWithoutChildren = Map.of(this.props)..remove('children');
+    final nextPropsWithoutChildren = Map.of(nextProps)..remove('children');
+
+    bool shouldUpdate = true;
+
+    try {
+      shouldUpdate = !_deepCollectionEquality.equals(currentPropsWithoutChildren, nextPropsWithoutChildren) ||
+          !_deepCollectionEquality.equals(this.state, nextState) ||
+          (supportsPropChildren && !const ListEquality().equals(this.props['children'], nextProps['children']));
```

Comment:
Similarly, I don't think we should deep-check children.

With https://github.com/Workiva/over_react/pull/462, empty lists of children will always be equal.

In other cases, if a component is being rerendered with a new List of children, it's likely that thee children themselves will be new ReactElement instances as well. If consumers want to optimize those cases, they can override `shouldComponentUpdate` with a custom implementation, or use memoization to prevent new `ReactElement` creation at their component's consumption-site, eliminating the need for `shouldComponentUpdate`.

Reply:
@greglittlefield-wf so are you saying you think we should just do a simple equality check on children? Or some other sort of shallow comparison?

Reply:
Yeah, I think just simple a simple `identical` check, treating it like other props (see above comment).